samtranslator/model/sam_resources.py

```diff
@@ -1923,6 +1947,8 @@ def _construct_lambda_permission_policy(
         source: ConnectorResourceReference,
         destination: ConnectorResourceReference,
         profile: ConnectorProfile,
+        index: int,
+        multi_dest: bool,
```

Comment:
Could have a single variable by e.g. using `index: Optional[int]` instead.

Reply:
the index is from the loop. So I think we still need `multi_dest` flag to decide it's a single destination or a list that only includes 1 destination.

Reply:
`index` is only used for multi-resource connectors right? An "index" doesn't make sense for a non-list structure anyway, so I don't see why it should be relevant when not using multi-resource.

Reply:
(Also might want to make that `dest_index` for clarity. There's a good change we'll do multi-source as well.)